src/v/cluster/tm_stm_types.h

```diff
@@ -14,17 +14,109 @@
+ *     |                      |         abort
+ *  commit                   add          |
+ *     |              partition/offset    |
+ *     |                      |           |
+ *     |                      |           |
+ *     |              +-------v-------+   |
+ *     |              |               |   |
+ *     |    +---------+    ongoing    +---+-----+
+ *     |    |         |               |   |     |
+ *     |    |         +-------+-------+   |
+ *     |    |                 |           |  producer
+ *     |  commit            abort   +-----+   fenced
+ *     |    |                 |     |           |
+ *     |    |                 |     |           |
+ *  +--v----v-------+ +-------v-----v-+ +-------v-------+
+ *  |               | |               | |               |
+ *  |prepare_commit | | prepare_abort | |    expired    |
+ *  |               | |               | |               |
+ *  +-------+-------+ +------+--------+ +------------+--+
+ *          |                |                       |
+ *    done committing   done aborting            done aborting
+ *          |                |                       |
+ * +--------v--------+       |  +----------------+   |
+ * |                 |       |  |                |   |
+ * | complete_commit |       +->| complete_abort |<--+
+ * |                 |          |                |
+ * +--------+--------+          +--------+-------+
+ *          |                            |
+ *      expired                       expired
+ *          |                            |
+ *          |      +---------------+     |
+ *          |      |               |     |
+ *          +----->|   tombstone   |<----+
+ *                 |               |
+ *                 +---------------+
+ */
 enum tx_status : int32_t {
-    ongoing,
-    preparing,
-    prepared,
-    aborting, // abort is initiated by a client
-    killed,   // abort is initiated by a timeout
-    ready,
-    tombstone,
+    /**
+     * When transactional id to producer id mapping is added to coordinator it
+     * starts in this state
+     */
+    empty = 5,
+    /**
+     * Transaction is ongoing as soon as partition or offset is added to it
+     */
+    ongoing = 0,
+    /**
+     * Transaction is marked as preparing_commit when it is requested to be
+     * committed by the client but before committing transaction on data and
+     * offset partitions
+     */
+    preparing_commit = 1,
+    /**
+     * Transaction is completed commit when its participants committed the
+     * transaction.
+     */
+    completed_commit = 2,
+    /**
+     * Transaction is marked as preparing_abort when it is requested to be
+     * aborted by the client but before aborting transaction on data and
+     * offset partitions
+     */
+    preparing_abort = 3,
+    /**
+     * The same as the preparing_abort but when transaction is aborted by
+     * internals of Redpanda f.e. timeout or transactional id limit overflow
+     */
+    expired = 4,
```

Comment:
`tx_gateway_frontend::expire_tx` moves a transaction to the `tombstone` state, not to `expired`, I think this may lead to confusion. Maybe this state should be named differently?

Reply:
agreed, i will rename it